inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -67,11 +67,15 @@ class XmlVisitor : public ngraph::AttributeVisitor {
     }
     void on_adapter(const std::string& name,
                     ngraph::ValueAccessor<std::string>& adapter) override {
-        // __generic_ie_type__ should not be serialized as a <data> attribute
-        // it is a WA to retrieve layer type name without introducing dependency on
-        // plugi_api library on transformations library
-        if (name == "__generic_ie_type__") {
-            ie_generic_type_name = adapter.get();
+        if ((m_node_type_name == "GenericIE") && (name == "__generic_ie_type__")) {
+            // __generic_ie_type__  in GenericIE should not be serialized as a <data>
+            // since it's purpose is to hold name of the layer type
+            // it is a WA to not introduce dependency on plugin_api library
+            m_node_type_name = adapter.get();
+        } else if ((m_node_type_name == "ExecutionNode") && (name == "layerType")) {
```

Comment:
Why cannot we use one if for these conditionals?

Reply:
Sure we can, thx for noticing - I must be tired today or something :)